.mvn/extensions.xml

```diff
@@ -0,0 +1,7 @@
+<extensions xmlns="http://maven.apache.org/EXTENSIONS/1.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/EXTENSIONS/1.0.0 http://maven.apache.org/xsd/core-extensions-1.0.0.xsd">
```

Comment:
not related should be part of an other pr

Reply:
can you please provide with a `CONTRIBUTING.md` on how you want your PR's to be split up.

Reply:
I think the general rule is one functional (independent) change = PR and it's for two reasons:
easier reviewing (geez, that PR is big!) and easier backward reference when tracking changes.
So here, definitely the first two bullet points seem to me as two separate independent PRs:
```
* Convert global configuration to an actual Global Configuration extension
* Populate jira sites from folders in Job configuration.
```
The rest looks like nice improvements to the coding style and code testability/readability and I guess they're fine to be together in one PR.

Reply:
Global configuration improvement went in because that made it easier for me to introduce support and testing for Jcasc 😅

Reply:
incremental build should be separate. 
well it's not a matter of having paperwork/procedure/etc... but as said @warden this make pr smaller to review and containing one change. So it's easier to revert if it breaks..